token/program-2022-test/tests/group_member_pointer.rs

```diff
@@ -0,0 +1,554 @@
+#![cfg(feature = "test-sbf")]
+
+mod program_test;
+use {
+    program_test::{TestContext, TokenContext},
+    solana_program::system_instruction,
+    solana_program_test::{
+        processor,
+        tokio::{self, sync::Mutex},
+        ProgramTest, ProgramTestContext,
+    },
+    solana_sdk::{
+        account::Account as SolanaAccount,
+        instruction::InstructionError,
+        pubkey::Pubkey,
+        signature::Signer,
+        signer::keypair::Keypair,
+        transaction::{Transaction, TransactionError},
+        transport::TransportError,
+    },
+    spl_token_2022::{
+        error::TokenError,
+        extension::{
+            group_member_pointer::{
+                instruction::{initialize, update},
+                GroupMemberPointer,
+            },
+            BaseStateWithExtensions, ExtensionType,
+        },
+        processor::Processor,
+        state::Mint,
+    },
+    spl_token_client::token::{ExtensionInitializationParams, TokenError as TokenClientError},
+    std::{convert::TryInto, sync::Arc},
+};
+
+fn setup_program_test() -> ProgramTest {
+    let mut program_test = ProgramTest::default();
+    program_test.prefer_bpf(false);
+    program_test.add_program(
+        "spl_token_2022",
+        spl_token_2022::id(),
+        processor!(Processor::process),
+    );
+    program_test
+}
+
+async fn setup_group_mint(
+    context: Arc<Mutex<ProgramTestContext>>,
+    mint: Keypair,
+    authority: &Pubkey,
+) -> TestContext {
+    let mut context = TestContext {
+        context,
+        token_context: None,
+    };
+    let group_address = Some(mint.pubkey());
+    context
+        .init_token_with_mint_keypair_and_freeze_authority(
+            mint,
+            vec![ExtensionInitializationParams::GroupPointer {
+                authority: Some(*authority),
+                group_address,
+            }],
+            None,
+        )
+        .await
+        .unwrap();
+    context
+}
+
+async fn setup_token_group(
+    token_context: &TokenContext,
+    mint_authority: &Keypair,
+    update_authority: &Pubkey,
+    payer: &Keypair,
+) {
+    token_context
+        .token
+        .token_group_initialize_with_rent_transfer(
+            &payer.pubkey(),
+            &mint_authority.pubkey(),
+            update_authority,
+            3,
+            &[&payer, &mint_authority],
+        )
+        .await
+        .unwrap();
+}
+
+async fn setup_member_mint(
+    context: Arc<Mutex<ProgramTestContext>>,
+    mint: Keypair,
+    group_address: &Pubkey,
+    authority: &Pubkey,
+    group_update_authority: &Keypair,
+) -> TestContext {
+    let mut context = TestContext {
+        context,
+        token_context: None,
+    };
+    let member_address = Some(mint.pubkey());
+    context
+        .init_token_with_mint_keypair_and_freeze_authority_and_additional_signers(
```

Comment:
I don't think we'll need this new function if the group member pointer extension looks like metadata pointer / group pointer